cmd/conduit/root/pipelines/describe_test.go

```diff
@@ -54,3 +64,127 @@ func TestDescribeExecutionCorrectArgs(t *testing.T) {
 	is.NoErr(err)
 	is.Equal(c.args.PipelineID, pipelineID)
 }
+
+func TestDescribeCommandExecuteWithClient(t *testing.T) {
+	is := is.New(t)
+
+	buf := new(bytes.Buffer)
+	out := &ecdysis.DefaultOutput{}
+	out.Output(buf, nil)
+
+	ctx := context.Background()
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	cmd := &DescribeCommand{args: DescribeArgs{PipelineID: "1"}}
+	cmd.Output(out)
+
+	mockPipelineService := mock.NewMockPipelineService(ctrl)
+	mockProcessorService := mock.NewMockProcessorService(ctrl)
+	mockConnectorService := mock.NewMockConnectorService(ctrl)
+
+	testutils.MockGetPipeline(mockPipelineService, cmd.args.PipelineID, []string{"conn1", "conn2"}, []string{"proc1", "proc2"})
+
+	testutils.MockGetProcessor(
+		mockProcessorService,
+		"proc1", "field.set", "",
+		nil,
+		map[string]string{
+			".Payload.After.department": "finance",
+		})
+
+	testutils.MockGetProcessor(
+		mockProcessorService,
+		"proc2", "custom.javascript", "",
+		nil,
+		map[string]string{})
+
+	testutils.MockGetConnectors(mockConnectorService, cmd.args.PipelineID, []*apiv1.Connector{
+		{Id: "conn1", Type: apiv1.Connector_TYPE_SOURCE, Plugin: "plugin1", ProcessorIds: []string{"proc3"}},
+		{Id: "conn2", Type: apiv1.Connector_TYPE_DESTINATION, Plugin: "plugin2"},
+	})
+
+	testutils.MockGetConnector(
+		mockConnectorService, "conn1", "plugin1", cmd.args.PipelineID, apiv1.Connector_TYPE_SOURCE,
+		&apiv1.Connector_Config{
+			Name: "Test Pipeline",
+			Settings: map[string]string{
+				"foo": "bar",
+			},
+		}, []string{"proc3"})
+
+	testutils.MockGetProcessor(
+		mockProcessorService,
+		"proc3", "custom.javascript", "",
+		nil,
+		map[string]string{})
+
+	testutils.MockGetConnector(
+		mockConnectorService, "conn2", "plugin2", cmd.args.PipelineID, apiv1.Connector_TYPE_DESTINATION,
+		&apiv1.Connector_Config{
+			Name: "Test Pipeline",
+			Settings: map[string]string{
+				"foo": "bar",
+			},
+		}, []string{"proc3"})
+
+	testutils.MockGetDLQ(mockPipelineService, cmd.args.PipelineID, "dlq-plugin")
+
+	client := &api.Client{
+		PipelineServiceClient:  mockPipelineService,
+		ProcessorServiceClient: mockProcessorService,
+		ConnectorServiceClient: mockConnectorService,
+	}
+
```

Comment:
Maybe it makes sense to have a helper method that creates this client and the needed mock services?

Reply:
@hariso that's a good idea. I think I'll ship it like this, and maybe re-iterate on this on future PRs. Thanks for the suggestion!